common/scripts/mesh-11s.sh

```diff
@@ -82,32 +109,116 @@ EOF
       (batadv-vis -i bat0 -s)&
       echo "started batadv-vis"
 
-      wpa_supplicant -i $wifidev -c /var/run/wpa_supplicant-11s.conf -D nl80211 -C /var/run/wpa_supplicant/ -B
+      # FIXME: Like the comment above - we need to figure out how to handle
+      # multiple Wi-Fi interfaces better. For some reason the background setting
+      # ensures wpa_supplicant doesn't start when this script is run as a process.
+      # This is likely due to the interface not being up in time, and will
+      # require some fiddling with the systemd startup order.
+      if [[ -z "${10}" ]]; then
+        wpa_supplicant -i "$wifidev" -c /var/run/wpa_supplicant-adhoc.conf -D nl80211 -C /var/run/wpa_supplicant/ -B
+      else
+        wpa_supplicant -i "$wifidev" -c /var/run/wpa_supplicant-adhoc.conf -D nl80211 -C /var/run/wpa_supplicant/
+      fi
       ;;
+
 ap)
       if [[ -z "$1" ]]
         then
           echo "check arguments..."
         help
       fi
 
-      # remove all saved wifi networks
-      nmcli --pretty --fields UUID,TYPE con show | grep wifi | cut -b -37 | while read line; do nmcli con delete uuid  $line; done
-
-      nmcli dev set $wifidev managed yes
-      nmcli radio wifi on
-
       # find ap parameters from enclave
       if [ -z "$DRONE_DEVICE_ID" ]
       then
-            echo "DRONE_DEVICE_ID not available"
-            nmcli device wifi hotspot con-name debug-hotspot ssid debug_hotspot band a channel 36 password 1234567890
+        echo "DRONE_DEVICE_ID not available"
+
+cat <<EOF >/var/run/wpa_supplicant-ap.conf
+ctrl_interface=DIR=/var/run/wpa_supplicant
+ap_scan=1
+p2p_disabled=1
+network={
+    ssid="debug-hotspot"
+    mode=2
+    frequency=5220
+    key_mgmt=WPA-PSK
```

Comment:
It would be nice to be able to pass a param to turn this off/on for testing purposes. For example, we may actually wish to have this unsecured for easy on-air monitoring, or to improve performance. Doesn't need to be fixed here but let's raise it as a github issue?